### PR TITLE
Support for torch.nested.as_nested_tensor(t)

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #121518
* __->__ #113280
* #113279

This PR adds support for tensor inputs to `as_nested_tensor()`. The tensor is treated as a batch of consistently-sized constituents. It utilizes `_nested_view_from_values_offsets()` to return a real view that allows for propagating gradients into inputs.

### CODE REVIEW

**Review by cpuhrsch — APPROVED** [review]

**Review by soulitzer — APPROVED** [review]

File: test/test_nestedtensor.py [hunk 1]
```diff
@@ -3317,6 +3317,59 @@ def test_jagged_view_from_values_offsets(self, device, dtype, requires_grad):
...
+        orig_t = t
+        if not contiguous:
+            t = t.transpose(0, 1)
+
+        nt = torch.nested.as_nested_tensor(t, device=device, dtype=dtype, layout=layout)
+        self._validate_nt(nt, list(t.unbind(0)), device, dtype, requires_grad, layout)
+
+        if torch.device(device) == t.device and dtype == t.dtype and contiguous:
+            # should be the non-copying (view) case
+            # TODO: Figure out why this is broken!
```
[the remark this answers is not in the document]
Comment (jbschlosser) [reply test/test_nestedtensor.py:3350]: nah, good catch. should be able to remove it now

File: torch/nested/__init__.py [hunk 2]
```diff
@@ -52,23 +61,59 @@ def as_nested_tensor(
...
-    ):
+    is_tensor_list = isinstance(ts, list) and all(isinstance(t, Tensor) for t in ts)
+    if not isinstance(ts, Tensor) and not is_tensor_list:
         raise TypeError(
-            "as_nested_tensor(): Expected first argument to be a list of tensors "
+            "as_nested_tensor(): Expected first argument to be a tensor or a list of tensors "
         )
 
+    if isinstance(ts, Tensor) and ts.dim() < 2:
+        raise RuntimeError("as_nested_tensor(): Expected tensor argument to have dim() > 1")
```
Comment (Skylion007) [inline torch/nested/__init__.py:74]:
```suggestion
        raise ValueError("as_nested_tensor(): Expected tensor argument to have dim() > 1")
```
